ray-operator/controllers/ray/rayjob_controller_test.go

```diff
@@ -289,6 +292,22 @@ var _ = Context("Inside the default namespace", func() {
 				time.Second*15, time.Millisecond*500).Should(Equal(rayv1.JobDeploymentStatusComplete), "jobDeploymentStatus = %v", myRayJob.Status.JobDeploymentStatus)
 			Expect(myRayJob.Status.EndTime.After(now)).Should(BeTrue(), "EndTime = %v, Now = %v", myRayJob.Status.EndTime, now)
 		})
+
+		It("job completed with ShutdownAfterJobFinishes=true, RayCluster should be deleted but not the submitter Job", func() {
```

Comment:
We need to split the test logic in the future instead of adding tests in a single `Context` block. All tests coupling together make us hard to maintain.